autobahn/wamp_session.ipp

```diff
@@ -936,14 +936,68 @@ inline void wamp_session::process_error(wamp_message&& message)
                     call_itr->second->result().set_exception(boost::copy_exception(std::runtime_error(error)));
                     m_calls.erase(call_itr);
                 } else {
-                    throw protocol_error("bogus ERROR message for non-pending CALL request ID");
+                    throw protocol_error("bogus ERROR message for non-pending CALL request ID: " + error);
+                }
+            }
+            break;
+        case message_type::REGISTER:
+            {
+                auto reg_itr = m_register_requests.find(request_id);
+                if (reg_itr != m_register_requests.end())
+                {
+                    reg_itr->second->response().set_exception(boost::copy_exception(std::runtime_error(error)));
+                    m_register_requests.erase(reg_itr);
+                } else {
+                    throw protocol_error("bogus ERROR message for non-pending REGISTER request ID: " + error);
+                }
+            }
+            break;
+        case message_type::UNREGISTER:
+            {
+                auto unreg_itr = m_unregister_requests.find(request_id);
+                if (unreg_itr != m_unregister_requests.end())
+                {
+                   unreg_itr->second->response().set_exception(boost::copy_exception(std::runtime_error(error)));
+                   m_unregister_requests.erase(unreg_itr);
+                } else {
+                    throw protocol_error("bogus ERROR message for non-pending UNREGISTER request ID: " + error);
+                }
+            }
+            break;
+        case message_type::PUBLISH:
+            {
+                //TODO: there's currently no way to get to the future returned by the publish function
```

Comment:
With unacknowledged publications, there is no future to put the error on. With acknowledged publications, there is a future, and we could stick any publication errors onto that.